<launch>

  <node pkg="hydro_system" type="DS18B20_node.py" name="water_temp_sensor" >
    <remap from="/temp" to="/water_temp" />
  </node>

  <!--
  <node pkg="hydro_system" type="filter_node.py" name="temp_filter" >
    <remap from="/raw_in" to="/water_temp" />
    <remap from="/filtered_out" to="/water_temp_filtered" />
    <param name="q" value="0.001" />
    <param name="r" value="0.1" />
    <param name="var" value="0.1" />
  </node>
  -->

  <node pkg="hydro_system" type="median_filter.py" name="temp_filter" >
    <remap from="/raw_in" to="/water_temp" />
    <remap from="/filtered_out" to="/water_temp_filtered" />
    <param name="width" value="60" />
  </node>

  <node pkg="hydro_system" type="pH_sensor_node.py" name="pH_sensor" output="screen" />

  <!--
  <node pkg="hydro_system" type="filter_node.py" name="pH_filter" >
    <remap from="/raw_in" to="/pH" />
    <remap from="/filtered_out" to="/pH_filtered" />
    <param name="q" value="0.001" />
    <param name="r" value="0.1" />
    <param name="var" value="0.1" />
    <param name="chi2" value="1.5" />
  </node>
  -->

  <node pkg="hydro_system" type="median_filter.py" name="pH_filter" >
    <remap from="/raw_in" to="/pH" />
    <remap from="/filtered_out" to="/pH_filtered" />
    <param name="width" value="60" />
  </node>

  <node pkg="hydro_system" type="SHT31D_node.py" name="air_temp_humidity">
    <remap from="/temp" to="/air_temp" />
  </node>

  <!--
  <node pkg="hydro_system" type="filter_node.py" name="air_temp_filter" >
    <remap from="/raw_in" to="/air_temp" />
    <remap from="/filtered_out" to="/air_temp_filtered" />
    <param name="q" value="0.001" />
    <param name="r" value="0.1" />
    <param name="var" value="0.1" />
  </node>

  <node pkg="hydro_system" type="filter_node.py" name="water_temp_filter" >
    <remap from="/raw_in" to="/humidity" />
    <remap from="/filtered_out" to="/humidity_filtered" />
    <param name="q" value="0.001" />
    <param name="r" value="0.1" />
    <param name="var" value="0.1" />
  </node>
  -->

  <node pkg="hydro_system" type="median_filter.py" name="air_temp_filter" >
    <remap from="/raw_in" to="/air_temp" />
    <remap from="/filtered_out" to="/air_temp_filtered" />
    <param name="width" value="30" />
  </node>

  <node pkg="hydro_system" type="median_filter.py" name="humidity_filter" >
    <remap from="/raw_in" to="/humidity" />
    <remap from="/filtered_out" to="/humidity_filtered" />
    <param name="width" value="30" />
  </node>

  <node pkg="hydro_system" type="motor_hat_node.py" name="motor_hat" />
  
  <node pkg="hydro_system" type="pH_controller_node.py" name="pH_controller" output="screen" >
    <param name="set_point" value="6.0" />
    <param name="range" value="0.2" />
    <param name="adjust_duration" value="1800" />
  </node> 

  <node pkg="hydro_system" type="logger_node.py" name="logger" >
    <param name="topics" value="/water_temp /water_temp_filtered /air_temp /air_temp_filtered /pH /pH_filtered /humidity /humidity_filtered" />
    <param name="period" value="60" />
  </node>

  <node pkg="hydro_system" type="timelapse_node.py" name="timelapse" >
    <param name="interval" value="600" />
  </node>

</launch>
